### ISSUE TITLE
Get connected client ip address @ server

### ISSUE THREAD

**Issue by borishardbass** [yhirose/cpp-httplib#39]
Need an easy way to access it, perhaps in request structure

**Comment by kevinbcarpenter** [comment]
I had to solves this problem and considered two solutions.  First was to capture the IP address at the accept call (in Server::listen), but then I would need to pass the IP into the request and the scope seemed wrong.  Alternatively I figured I could write a function for SocketStream and get the IP from there and set it in the request.  The + notes my additions.

```
 class Stream {

 public:
     virtual ~Stream() {}
     virtual int read(char* ptr, size_t size) = 0;
     virtual int write(const char* ptr, size_t size1) = 0;
     virtual int write(const char* ptr) = 0;
+    virtual std::string get_remote_addr() = 0;

 
     template <typename ...Args>
     void write_format(const char* fmt, const Args& ...args);
 };

 
 class SocketStream : public Stream {
 public:
     SocketStream(socket_t sock);
     virtual ~SocketStream();

     virtual int read(char* ptr, size_t size);
     virtual int write(const char* ptr, size_t size);
     virtual int write(const char* ptr);

+    std::string get_remote_addr() {
+        socklen_t len;
+        struct sockaddr_storage addr;
+        char ipstr[INET6_ADDRSTRLEN];
+
+        len = sizeof addr;
+        getpeername(sock_, (struct sockaddr*)&addr, &len);
+
+        // deal with both IPv4 and IPv6:
+        if (addr.ss_family == AF_INET) {
+            struct sockaddr_in *s = (struct sockaddr_in *)&addr;
+            inet_ntop(AF_INET, &s->sin_addr, ipstr, sizeof ipstr);
+        } else { // AF_INET6
+            struct sockaddr_in6 *s = (struct sockaddr_in6 *)&addr;
+            inet_ntop(AF_INET6, &s->sin6_addr, ipstr, sizeof ipstr);
+        }
+
+        return ipstr;
+    }
+
 private:
     socket_t sock_;
 };
```

Then in Server::process_request I added this line:
```
+     req.set_header("REMOTE_ADDR", strm.get_remote_addr().c_str());
```

**Comment by yhirose** [comment]
Closed with #41.
